add new spec to references pages (#6190)
fishtown-analytics/docs.getdbt.com: website/docs/reference/resource-configs/unique_key.md

datatype: column_name_or_expression
---

<VersionBlock firstVersion="1.9">

<File name='snapshots/<filename>.yml'>

```yaml
snapshots:
  - name: orders_snapshot
    relation: source('my_source', 'my_table')
    [config](/reference/snapshot-configs):
      unique_key: id

```

</File>
</VersionBlock>

<VersionBlock lastVersion="1.8">

:::info Use the latest snapshot syntax

In Versionless and dbt v1.9 and later, snapshots are defined in an updated syntax using a YAML file within your `snapshots/` directory (as defined by the [`snapshot-paths` config](/reference/project-configs/snapshot-paths)). For faster and more efficient management, consider the updated snapshot YAML syntax, [available in Versionless](/docs/dbt-versions/versionless-cloud) or [dbt Core v1.9 and later](/docs/dbt-versions/core).

> :::

Reviewer: We've got a couple of these -- would we ever want a snippet for this info call out?

```

</File>
</VersionBlock>

Though, it's probably a better idea to construct this column in your query and use that as the `unique_key`:

<VersionBlock firstVersion="1.9">

<File name='snapshots/transaction_items_snapshot.yml'>

```yaml
snapshots:
  - name: transaction_items_snapshot
    relation: {{ ref('transaction_items_ephemeral') }}
    config:
      schema: snapshots
      unique_key: id
      strategy: timestamp
      updated_at: updated_at

# models/transaction_items_ephemeral.sql
{{ config(materialized='ephemeral') }}

select
  transaction_id || '-' || line_item_id as id,
  *
from {{ source('erp', 'transactions') }}

```

> </File>

Reviewer: It looks like the intent was two different file examples here. So I'd suggest one extra pair of `<File ...>...</File>` get added in just the right spot.

> In this example, we create an ephemeral model `transaction_items_ephemeral` that creates the unique key id, and then references it in our snapshot.

Reviewer: This is a little more clear to me:
```suggestion
In this example, we create an ephemeral model `transaction_items_ephemeral` that creates an `id` column that can be used as the `unique_key` our snapshot configuration.
```
